iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/compaction/IcebergQueryCompactor.java

```diff
@@ -62,20 +63,32 @@ public boolean run(CompactorContext context) throws IOException, HiveException,
     Table icebergTable = IcebergTableUtil.getTable(conf, table.getTTable());
     String compactionQuery;
     String orderBy = ci.orderByClause == null ? "" : ci.orderByClause;
+    String fileSizeCond = null;
+
+    if (ci.type == CompactionType.MINOR) {
```

Comment:
so it's not worth splitting into individual class strategies for clarify?

Reply:
I modified this piece of code and there is no difference in code between Minor and Major compaction in the current implementation. The difference is in behavior, minor compaction can be done when there are no files with deletes and it compacts smaller files than major compaction and it is faster than major compaction.